cpp/ql/lib/semmle/code/cpp/Variable.qll

```diff
@@ -410,6 +410,10 @@ class LocalVariable extends LocalScopeVariable, @localvariable {
     or
     orphaned_variables(underlyingElement(this), unresolveElement(result))
   }
+
+  override predicate isStatic() {
+    super.isStatic() or orphaned_variables(underlyingElement(this), _)
```

Comment:
Is it safe to assume that all orphaned variables are static? Or should we also compare the name to some allow list in the case the variable is an orphan?

Reply:
It's safe according to the frontend documentation.

Reply:
They must be static, because they originate from function scope, but can be referred to from outside the function.